### DISCUSSION TITLE
blog/global-exception-handling-in-aspnet-core/

### DISCUSSION THREAD

**Discussion by giscus** [codewithmukesh/discussions#7] [General]
# blog/global-exception-handling-in-aspnet-core/

Learn about Global Exception Handling in ASP.NET Core applications (.NET 8). Exceptions are something inevitable in any product and should be handled as effectively as possible without breaking the system. This can occur due to both internal as well as external factors. We will mainly discuss about IExceptionHandler, which was recently introduced starting from .NET 8. This is a far cleaner and maintainable approach while working with exceptions.

https://codewithmukesh.com/blog/global-exception-handling-in-aspnet-core/

**Comment by dkj468** [comment]
@iammukeshm  , What is the advantage of using this newly introduced feature over previous method which is error handling middleware ? Just trying to understand why DOTNET team has introduced this feature ?

**Comment by zabronm** [comment]
Way you write some of your articles seems is for the experts who would normally already be aware of the subject. If you could explain in your code where to put what procedure explicitly rather than assuming. I got lost too early and abandoned rest of the article. Please do not assume all your audiences know all the concepts that you would normally call basics.

**Comment by iago-mota-dev** [comment]
Really good! Keep with the quality content, helped me a lot to understand new ways of error handling in .net.
For those complaning in the comments, i just recommend to keep studying, cause you need.

**Comment by CalebTek** [comment]
This is a very nice article on handling Exception

**Comment by MarcoJST** [comment]
Great article, thank you!
I generally find that built-in .NET exceptions are enough for my scenarios, which helps avoid adding extra code to projects. Don't you think that always creating custom exceptions could be a little bit overkill for bigger projects? Do you have any recommendations on best practices for this?
Another question that occurs to me is whether with this general exception handling there is a risk that an unhandled exception might end up sending sensitive data to the client, right? What are your recommendations for this kind of scenarios?

**Comment by mohamedHodaib** [comment]
Keep Going to get this course Done  ❤❤
